### PR TITLE
Fixing gcc/clang warnings

### PR DESCRIPTION
**Pull request by AnkeAnke**
Mostly unused variable and redundant move.
Export of specialized template classes failed on clang (silently) gcc.
Tested on gcc-7/Ubuntu 20 (Pop! OS) and MSVC 14/Windows 10 (and clang/jenkins).

### CODE REVIEW

**Review by petersteneteg — DISMISSED** [review]

File: modules/dataframe/src/datastructures/dataframe.cpp [hunk 1]
```diff
@@ -161,6 +161,19 @@ DataFrame::DataFrame(const DataFrame &df) {
     }
 }
 
+DataFrame &DataFrame::operator=(const DataFrame &df) {
+    for (const auto &col : df.columns_) {
```
Comment (r-englund) [inline modules/dataframe/src/datastructures/dataframe.cpp]: Need to `clear` `columns_` on `this` first. 
Also, need to check such that `this != *df`

File: modules/dataframe/src/datastructures/dataframe.cpp [hunk 2]
```diff
@@ -161,6 +161,19 @@ DataFrame::DataFrame(const DataFrame &df) {
...
+        columns_.emplace_back(col->clone());
+    }
+    return *this;
+}
+
+DataFrame &DataFrame::operator=(const DataFrame &&df) {
```
Comment (r-englund) [inline modules/dataframe/src/datastructures/dataframe.cpp]: move ctor and assign-op take non-const input or else you cant move out data from the rhs

Comment (r-englund) [reply modules/dataframe/src/datastructures/dataframe.cpp]: Also, I think the move ctor and assign-op could use `= default`

**Review by r-englund — CHANGES_REQUESTED** [review]

File: modules/dataframe/src/datastructures/dataframe.cpp [hunk 3]
```diff
@@ -162,6 +162,8 @@ DataFrame::DataFrame(const DataFrame &df) {
 }
 
 DataFrame &DataFrame::operator=(const DataFrame &df) {
+    if (this == &df) return;
```
Comment (petersteneteg) [inline modules/dataframe/src/datastructures/dataframe.cpp]: should always return *this...

File: modules/dataframe/src/datastructures/dataframe.cpp (hunk 2 above)
Comment (AnkeAnke) [reply modules/dataframe/src/datastructures/dataframe.cpp]: Fixed the first. Movign the std vector is more efficient than copying, thus the move overrides.

**Review by r-englund — DISMISSED** [review]

**Review by petersteneteg — APPROVED** [review]

**Pull request merged**